module/input_handler.py

```diff
@@ -1,56 +1,43 @@
-from module.command_node import NodeType
-from module.frame_functions import cast_and_send_ui_frame
+from module.command_node import Node, Command
 
 
 class input_handler:
     def __init__(self, cli_controller):
         self.cli_controller = cli_controller
 
     @staticmethod
-    def print_command_not_found(command: str) -> None:
+    def _print_command_not_found(command: str) -> None:
         """
         Prints command not found text with the command parameter.
         """
-        print("\tCommand '{}' not found, type 'help' for possible commands.".format(command))
+        print(
+            "\tCommand '{}' not found, type 'help' for possible commands.".format(
+                command
+            )
+        )
 
     @staticmethod
-    def check_amount_parameters(parameters: list, required_amount: int):
+    def _check_amount_parameters(parameters: list, required_amount: int) -> bool:
         """
         compares the length of the parameters list to the required amount,
         returns true when the length of the parameters is equal to the required amount
         prints to user if not enough or too many parameters
         """
         if len(parameters) < required_amount or len(parameters) > required_amount:
-            print("\tExpected {} parameters, got {}.".format(required_amount, len(parameters)))
+            print(
+                "\tExpected {} parameters, got {}.".format(
+                    required_amount, len(parameters)
+                )
+            )
             return False
         return True
 
-    def handle_help(self, help_parameters):
-        try:
-            node = self.cli_controller.current_node
-
-            for param in help_parameters:
-                node = node[param.upper()]
-
-            self.cli_controller.print_help(node)
-        except IndexError:
-            self.cli_controller.print_help(self.cli_controller.current_node)
-        except KeyError:
-            self.print_command_not_found(param)
-
-    def handle_select(self, select_parameters) -> bool:
-        if not self.check_amount_parameters(select_parameters, 1):
-            return False
-        if not select_parameters[0]:
-            print("\tInvalid parameter '{}'".format(select_parameters[0]))
-            return False
-
-        self.cli_controller.set_target(select_parameters[0])
-        print("\tYou selected {}".format(select_parameters[0]))
-        return True
-
     @staticmethod
-    def convert_type(convertable, convert_type):
+    def _convert_type(convertable: str, convert_type: type):
```

Comment:
Please be consistent and typehint the return type.

Reply:
The tricky part here about the return type is that the return type will be the type of `convert_type`, this has been clarified in the comments in the upcoming patch.

Reply:
Then you could use -> Any